cpmonitor/views.py

```diff
@@ -41,20 +50,30 @@ def _calculate_summary(request, node):
         subtasks = subtasks.filter(draft_mode=False)
 
     subtasks_count = len(subtasks)
-    status_counts = Counter([s.execution_status for s in subtasks])
+    status_counts = Counter(
+        [TO_STATUS[subtask.execution_status] for subtask in subtasks]
+    )
     status_proportions = {
-        s: round(c / subtasks_count * 100) for s, c in status_counts.items()
+        status: round(count / subtasks_count * 100)
+        for status, count in status_counts.items()
     }
 
-    node.status_proportions = [
-        (v, statuses[k].label, statuses[k].name)
-        for k, v in sorted(status_proportions.items(), reverse=True)
-    ]
+    node.status_proportions = _sort_status_proportions(status_proportions, STATUS_ORDER)
```

Comment:
I find it helpful that it has a name - even if it is just one line. Using `sort` with a lambda could simplify `_sort_status_proportions`, but this has some differences (type signature, status_counts instead of status_proportions).

Reply:
My mistake. status_counts should read status_proportions. Other than that it's identical, just shorter and IMHO more pythonic than using a "manual" loop. But no big deal either.